.github/workflows/zulip-tests.yml

```diff
@@ -0,0 +1,62 @@
+name: Build
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+
+  static-analysis:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+
+    - name: Install dependencies
+      run: |
+        sudo rm -vf /etc/apt/sources.list.d/*
+        sudo apt-get update
+        sudo apt-get install virtualenv
+        tools/provision --force
+
+    - name: Running Test-Suite
+      run: |
+        source zulip-api-py3-venv/bin/activate
+        tools/test-static-analysis
```

Comment:
I think it would be more efficient if you do this inside test, but somehow use `if` ( https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif ) to make sure it runs only on 3.6.

Reply:
This is a great idea -- it'll save the time of setting up python and installing deps an extra time.

Reply:
Can it generate a separate output, though? "3.6 & static checks" is not ideal, even if it does reduce setup, whereas if it could be run in the 3.6 section but generate a static test output section, that would be clearer for devs to read - particularly new ones.

Reply:
The step is going to be named "Static analysis" so I think it will be unambiguous for devs to read the nature of the error.

Reply:
I was wondering as well to have the static tests on every python version and run the rest only if it succeeds. @neiljp I am not sure of what you are talking about is possible, but I think having it on all the tests will make more sense.

Reply:
Though Having a separate Job for lint is good as well, because the jobs run in parallel and won't take more time. Rather Adding it as a step with increase the time of execution. I don't think we need to take care of its load of execution in the MS Servers.

Reply:
I suppose my concern was a relic of having to deal with Travis' limited resource (only 4 jobs max!). At this point it becomes a matter of whether being considerate to MSFT as not to waste their resource or not.

Reply:
Actually, this will help us find a separate report for the Static tests.

Reply:
@rht I agree that it would show the Step failed as Static Analysis, but it would show overall (only) within the named python 3.6 test job, not a specific Static Analysis job. That was my point about whether the result could be saved into a separate job as simply a result.

In zulip-terminal we currently have separate jobs for each part of static testing even, to make this clearer - or hopefully so - without too much overhead. Our installation dependencies and setup for each are lower though?

Reply:
I have no strong opinion on this separation of static analysis, so am fine with making it separate. Going forward, maybe it is worth it to cache pip to save time and resource during installation? See e.g. https://github.com/django-ftl/fluent-compiler/blob/a6048b856a3db0761ae05a9d3049489f6bb11cbe/.github/workflows/pythonpackage.yml#L29